lib/matplotlib/gridspec.py

```diff
@@ -49,9 +49,9 @@ def __init__(self, nrows, ncols, height_ratios=None, width_ratios=None):
 
     def __repr__(self):
         height_arg = (', height_ratios=%r' % (self._row_height_ratios,)
-                      if self._row_height_ratios is not None else '')
+                      if len({*self._row_height_ratios}) != 1 else '')
```

Comment:
Slightly prefer
```suggestion
                      if len(set(self._row_height_ratios)) != 1 else '')
```
for better readability (also below).

Reply:
I think I already won this battle: there's multiple occurrences of `{*foo}` in the codebase already and none of `set(*foo)` :-)

Reply:
Um, `set(*foo)` is not valid. It's `set(foo)` because `set()` takes a single iterable as argument. 

That's part of why I prefer it here: "make as set from this list" is just a single concept, whereas "unpack this iterable and create a set from its elements" is two concepts. (I don't have that much brain capacity https://youtu.be/UANN2Eu6ZnM :smile:).

Yes, you've already sneaked 5 times `{*foo}` into the code base. - I hope I didn't approve these :stuck_out_tongue_closed_eyes:.

But for real, `{*a, *b}` makes sense for multiple iterables, but for a single iterable `set(a)` is more readable than `{*a}`.

Reply:
sure, changed to set().